src/components/TaskPreviewPanel.js

```diff
@@ -1,68 +1,57 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { injectIntl } from 'react-intl';
+import { withTheme, withStyles } from '@material-ui/core/styles';
 import { Paper, Typography } from '@material-ui/core';
-import { makeStyles } from '@material-ui/styles';
-import TaskPreview from './TaskPreviewTable';
+import {
+  Contributions,
+  withHistory,
+  withModulesManager,
+} from '@openimis/fe-core';
+import { BENEFIT_PLAN_TASK_PREVIEW_TABLE_VALUE, TASKS_PREVIEW_CONTRIBUTION_KEY } from '../constants';
 
-const useStyles = makeStyles((theme) => ({
+const styles = (theme) => ({
   paper: theme.paper.paper,
   title: theme.paper.title,
-}));
+});
 
-const DUMMY_PREVIEW_ITEMS = [
+function TaskPreviewPanel({
+  intl,
+  classes,
+  rights,
+  formatMessage,
+  edited,
+}) {
+  const [taskTable, setTaskTable] = useState('');
+  const task = { ...edited };
 
-  {
-    id: 1,
-    source: 'Phone',
-    type: 'Support',
-    entity: 'Customer A',
-    assignee: 'John Doe',
-    businessStatus: 'Pending',
-    status: 'RECEIVED',
-    historic: {
-      id: 1,
-      source: 'Email',
-      type: 'Support',
-      entity: 'Customer A',
-      assignee: 'John Doe',
-      businessStatus: 'Pending',
-      status: 'RECEIVED',
-    },
-  },
-  {
-    id: 2,
-    source: 'Phone',
-    type: 'Sales',
-    entity: 'Customer B',
-    assignee: 'Jane Smith',
-    businessStatus: 'Completed',
-    status: 'CLOSED',
-    historic: {
-      id: 2,
-      source: 'Phone',
-      type: 'Sales',
-      entity: 'Customer B',
-      assignee: 'Jane Smith',
-      businessStatus: 'In Progress',
-      status: 'ONGOING',
-    },
-  },
-];
+  useEffect(() => {
+    switch (task?.source) {
+      case 'Benefit Plan':
```

Comment:
Could be better to use here const instead of row string. Also as we discussed maybe 'Benefit Plan Update' could be a better name.

Reply:
done